packages/contract-sources/index.js

```diff
@@ -33,3 +35,27 @@ module.exports = (pattern, callback) => {
       }
     });
 };
+
+const resolveSource = function(importPath, importedFrom, options) {
+  const SOURCES = sources(options);
+
+  for (let source in SOURCES) {
+    const { body, filePath } = source.resolve(importPath, importedFrom);
+    if (body) {
+      return {
+        resolvedBody: body,
+        resolvedPath: filePath,
+        currentSource: source
+      };
+    }
+  }
+
+  let message = "Could not find " + importPath + " from any sources";
+  if (importedFrom) message += "; imported from " + importedFrom;
+  throw new Error(message);
+}
+
+module.exports = {
```

Comment:
Instead of returning a collection of functions here, I think we should follow the pattern of the Resolver and use a class, then attaching it to the config.